File: src/main/java/no/novari/flyt/egrunnerverv/gateway/exception/NoSuchLinkException.java
package no.novari.flyt.egrunnerverv.gateway.exception;

import no.fint.model.resource.FintLinks;

public class NoSuchLinkException extends RuntimeException {

    public static NoSuchLinkException noSelfLink(FintLinks resource) {
        return new NoSuchLinkException(String.format("No self link in resource=%s", resource.toString()));
    }

    public NoSuchLinkException(String message) {
        super(message);
    }
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/exception/NoSuchLinkException.kt
package no.novari.flyt.egrunnerverv.gateway.exception

import no.novari.fint.model.resource.FintLinks

class NoSuchLinkException(
    message: String,
) : RuntimeException(message) {
    companion object {
        fun noSelfLink(resource: FintLinks): NoSuchLinkException =
            NoSuchLinkException("No self link in resource=$resource")
    }
}
